src/main/java/org/hobbit/core/components/channel/ReadByteChannel.java

```diff
@@ -0,0 +1,86 @@
+package org.hobbit.core.components.channel;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import com.rabbitmq.client.AMQP.BasicProperties;
+/**
+ * For each readByte call for DirectChannel implementation,
+ * a thread of ReadByteChannel is created. This makes sure that
+ * the execution takes place on a separate thread by waiting for 
+ * messages to be received by the pipe.
+ * @author altaf, sourabh, yamini, melisa
+ *
+ */
+public class ReadByteChannel extends DirectChannel implements Runnable{
+
+	PipeChannel pipeChannel;
+	DirectCallback callback;
+	public static ArrayList<Object> classes = new ArrayList<>();
+	private ExecutorService threadPool = Executors.newCachedThreadPool();
+
+	public ReadByteChannel(PipeChannel pipeChannel, Object callback, Object classs) {
+		this.pipeChannel = pipeChannel;
+		this.callback = (DirectCallback) callback;
+        classes.add(classs);
+	}
+
+	@Override
+	public void run() {
+		ByteBuffer buffer = ByteBuffer.allocate(1024);
+        try {
+        	if(pipeChannel != null && pipeChannel.getPipe() !=null) {
+        		while(pipeChannel.getPipe().source().isOpen() && 
+        				pipeChannel.getPipe().source().read(buffer) > 0){
+        			threadPool.execute(new ProcessCallback(callback, clone(buffer), pipeChannel.getProps()));
+        			buffer.clear();
```

Comment:
Wouldn't it be easier to execute `buffer = ByteBuffer.allocate(1024);` instead of cloning the buffer before clearing it?